<launch>
  <!-- subscribes to cmd_vel topic -->
  <!-- publishes odom topic with base_link frame -->
  <!-- broadcasts odom->base_link transform -->
  <arg name="param" default="value" />
  <node pkg="roboteq_diff_driver" type="driver" name="roboteq_diff_driver" output="screen">
    <!-- enable broadcast of odom tf -->
    <param name="pub_odom_tf" value="true" />
    <!-- specify odom frame -->
    <param name="odom_frame" value="odom" />
    <!-- specify base frame -->
    <param name="base_frame" value="base_link" />
    <!-- specify cmd_vel topic -->
    <param name="cmdvel_topic" value="cmd_vel" />
    <!-- specify port for roboteq controller -->
    <param name="port" value="/dev/roboteq_base" />
    <!-- specify baud for roboteq controller -->
    <param name="baud" value="115200" />
    <!-- specify whether to use open-loop motor speed control (as opposed to closed-loop)-->
    <param name="open_loop" value="false" />
    <!-- specify robot wheel circumference in meters  -->
    <param name="wheel_circumference" value="0.6283" /> <!--6283 -->

    <!-- 0.1m to actual 0.188m -->
    <!-- specify robot track width in meters  -->
    <param name="track_width" value="0.515" />
    <!-- specify pulse-per-revolution for motor encoders  -->
    <param name="encoder_ppr" value="2048" />
    <!-- specify counts-per-revolution for motor encoders (ppr*4 for quadrature encoders)  -->
    <param name="encoder_cpr" value="8192" />
    <!-- specify maximum motor current (amps) -->
    <param name="max_amps" value="50" />
    <!-- specify maximum motor speed (rpm) -->
    <param name="max_rpm" value="480" />
  </node>
</launch>
